navigationrail/navigationrail/src/main/java/com/microsoft/device/dualscreen/navigationrail/NavigationRailView.kt

```diff
@@ -0,0 +1,522 @@
+                super.onSwipeTop()
+                if (allowFlingGesture) {
+                    menuGravity = Gravity.TOP
+                }
+            }
+
+            override fun onSwipeBottom() {
+                super.onSwipeBottom()
+                if (allowFlingGesture) {
+                    menuGravity = Gravity.BOTTOM
+                }
+            }
+        }
+
+    /**
+     * Use an animation to move the buttons when the menu gravity is changed  or [arrangeButtons] is called.
+     * By default the [AccelerateDecelerateInterpolator] is used.
+     */
+    var useAnimation: Boolean = true
+
+    /**
+     * Set the interpolator for the animation when menu gravity is changed or [arrangeButtons] is called.
+     * By default the [AccelerateDecelerateInterpolator] is used.
+     */
+    var animationInterpolator: BaseInterpolator = AccelerateDecelerateInterpolator()
+
+    /**
+     * Allows the buttons to be moved to [DisplayPosition.START] or [DisplayPosition.END] with a swipe gesture.
+     */
+    var allowFlingGesture: Boolean = true
+
+    private fun extractAttributes(context: Context, attrs: AttributeSet?) {
+        val styledAttributes =
+            context.theme.obtainStyledAttributes(
+                attrs,
+                R.styleable.ScreenManagerAttrs,
+                0,
+                0
+            )
+        try {
+            screenMode = ScreenMode.fromId(
+                styledAttributes.getResourceId(
+                    R.styleable.ScreenManagerAttrs_tools_application_mode,
+                    ScreenMode.DUAL_SCREEN.ordinal
+                )
+            )
+            useAnimation =
+                styledAttributes.getBoolean(
+                    R.styleable.ScreenManagerAttrs_useAnimation,
+                    useAnimation
+                )
+            allowFlingGesture =
+                styledAttributes.getBoolean(
+                    R.styleable.ScreenManagerAttrs_allowFlingGesture,
+                    allowFlingGesture
+                )
+        } finally {
+            styledAttributes.recycle()
+        }
+    }
+
+    private var topScreenHeight = -1
+    private var bottomScreenHeight = -1
+    private var appWindowFrameHeight = -1
+    private var hingeHeight = -1
+    private var statusBarHeight = -1
+
+    private var screenSize = Point()
+    private var appWindowPosition = Rect()
+    private var hingePosition = Rect()
+    private var shouldRedrawMenu = true
+
+    private fun setScreenParameters() {
+        if (!isIntersectingHorizontalHinge()) {
+            return
+        }
+        context.getWindowVisibleDisplayFrame().let { windowRect ->
+            appWindowPosition = windowRect
+            appWindowFrameHeight = windowRect.height()
+
+            normalizeFoldingFeatureRectForView().let { hingeRect ->
+                hingePosition = hingeRect
+                (context as Activity).windowManager.defaultDisplay.getSize(screenSize)
```

Comment:
I think that you can use WindowMetricsCalculator from window manager library. I saw it that most of the functions from Activity.windowManager are deprecated and are not working verry well on our emulators.

Reply:
Done